terraform/rustc-ci/impl/artifacts.tf

```diff
@@ -148,3 +149,67 @@ resource "aws_s3_bucket_inventory" "artifacts" {
     }
   }
 }
+
+resource "aws_iam_role" "try_builds" {
+  name = "${var.iam_prefix}--try-role"
+
+  assume_role_policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Effect = "Allow"
+        Action = "sts:AssumeRoleWithWebIdentity"
+        Principal = {
+          Federated = "arn:aws:iam::890664054962:oidc-provider/token.actions.githubusercontent.com"
```

Comment:
Nit: I'd create a data source for the provider:

```terraform
data "aws_iam_openid_connect_provider" "gha" {
  url = "https://token.actions.githubusercontent.com"
}
```

...and get the ARN from there. That way we don't hardcode the account id.